src/ModelTable/ModelTable.tsx

```diff
@@ -84,7 +91,13 @@ export const ModelTable = Object.assign(
                 typeof scrollable === 'object' ? scrollable?.className : ''
               }
             >
-              <Table columnIds={tableColumns} data={data} {...tableProps}>
+              <Table
+                ref={ref}
+                columnIds={tableColumns}
+                data={data}
+                className={cn(rendered || 'w-full')}
```

Comment:
Use `&&` as its more explicit that it needs to be `!rendered`; Also have you tried to not use a rendered state and just use `!ref.current && 'w-full'` instead

Reply:
If we use `&&`, you will only apply `w-full` and try to stretch the columns after the render is complete, which does not work as it needs to be done before the render. Using `ref.current` instead of `rendered` could work, I can give it a try.

Reply:
Just changed to `!rendered && w-full`. Using `ref.current` results in some weird interactions, and my guess is because resizing causes ref.current to change (and possibly be undefined for a moment) and causing the `w-full` to be applied sporadically. However, `useEffect` is only activated when the table is first rendered, which is the interaction we want.